data-prepper-plugins/opensearch/src/test/java/com/amazon/dataprepper/plugins/sink/opensearch/index/DefaultIndexManagerTests.java

```diff
@@ -80,66 +88,200 @@ public void setup() throws IOException {
         when(indexConfiguration.getIsmPolicyFile()).thenReturn(Optional.empty());
-
-        defaultIndexManager = indexManagerFactory.getIndexManager(IndexType.CUSTOM, restHighLevelClient, openSearchSinkConfiguration);
-
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS);
         when(restHighLevelClient.cluster()).thenReturn(cluster);
         when(cluster.getSettings(any(ClusterGetSettingsRequest.class), any(RequestOptions.class)))
                 .thenReturn(clusterGetSettingsResponse);
 
-        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS);
         when(restHighLevelClient.indices()).thenReturn(indicesClient);
     }
 
+    @Test
+    public void getIndexAlias_IndexWithTimePattern(){
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS_WITH_TIME_PATTERN);
+        defaultIndexManager = indexManagerFactory.getIndexManager(IndexType.CUSTOM, restHighLevelClient, openSearchSinkConfiguration);
+        final String indexAlias = defaultIndexManager.getIndexAlias();
+        assertTrue(EXPECTED_INDEX_PATTERN.matcher(indexAlias).matches());
+        verify(openSearchSinkConfiguration, times(2)).getIndexConfiguration();
+        verify(indexConfiguration).getIndexAlias();
+        verify(indexConfiguration).getIsmPolicyFile();
+    }
+
+    @Test
+    public void getIndexAlias_IndexWithTimePattern_Exceptional_NotAsSuffix() {
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS_WITH_TIME_PATTERN + "randomtext");
+        assertThrows(IllegalArgumentException.class,
+                () -> indexManagerFactory.getIndexManager(IndexType.CUSTOM, restHighLevelClient, openSearchSinkConfiguration));
+        verify(openSearchSinkConfiguration).getIndexConfiguration();
+        verify(indexConfiguration).getIndexAlias();
+    }
+
+    private static final List<Character> INVALID_CHARS = Arrays.asList('#', '\\', '/', '*', '?', '"', '<', '>', '|', ',', ':');
+    @Test
+    public void getIndexAlias_IndexWithTimePattern_Exceptional_WithSpecialChars() {
+        INVALID_CHARS.stream().forEach(this::testIndexTimePattern_Exceptional_WithSpecialChars);
+        verify(openSearchSinkConfiguration, times(INVALID_CHARS.size())).getIndexConfiguration();
+        verify(indexConfiguration, times(INVALID_CHARS.size())).getIndexAlias();
+    }
+
+    private void testIndexTimePattern_Exceptional_WithSpecialChars(final Character character){
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS + "-%{yyyy" + character + ".MM.dd.HH}");
+        assertThrows(IllegalArgumentException.class,
+                () -> indexManagerFactory.getIndexManager(IndexType.CUSTOM, restHighLevelClient, openSearchSinkConfiguration));
+    }
+
+    @Test
+    public void testIndexTimePattern_Exceptional_MultipleTimePatterns(){
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS + "-%{yyyy}-{MM.dd.HH}");
```

Comment:
Should `"-%{yyyy}-{MM.dd.HH}"` be `"-%{yyyy}-%{MM.dd.HH}"`

Reply:
👍  done!

Reply:
👍  Done.